### PR TITLE
feat(cli): add hidden suppressions migrate gcp command

### PR DESCRIPTION
**Pull request by rmoles**
Signed-off-by: Ross <ross.moles@lacework.net>

## Summary

As part of the legacy compliance deprecation effort, the legacy cfg-analyzers will be decommissioned some time in January 2023. As a result customers will be moved across to the new LPP policies and CIS benchmarks. This is likely to cause a lot of noise for customers as they will have suppressions in place for resources which they expect to be in a non compliant state.
At present there is no easy path for customers to migrate their old suppressions to the new exceptions policies.

We do have a scripted approach created by Marc Garcia: https://github.com/lacework-dev/scripts/tree/main/lw-suppressions-migration

This PR adds the following hidden commands:

```
lacework suppressions migrate gcp
```
suppressions has the following alias' `suppression, sup, sups`
migrate has the following alias `mig`

`lacework suppressions migrate gcp`
Will enable users to migrate their legacy suppressions to new LPP policy Exceptions.
This supports the migration in 2 modes:
 - Manual (recommended): This outputs the converted suppressions as commands that the user can review and run as they wish
 - Automatic: This gives the user the option to allow Lacework to handle the suppressions migration. It will output the converted legacy suppressions in the Exception format and ask the user to confirm they are happy with the Exceptions and accepting liability for any Compliance violations that may be missed as a result of this.

Additionally this PR includes a fix for the `lacework suppressions migrate aws` command. This checks legacy suppressions for ARNs in the resourceNames field and parses out the resource name from the ARN, as ARNs are not supported for policy exceptions.

## How did you test this change?
Tested locally ✅ 

## Issue
https://lacework.atlassian.net/browse/RAIN-46613
https://lacework.atlassian.net/browse/RAIN-47184

### CODE REVIEW

File: cli/cmd/suppressions.go [hunk 1]
```diff
@@ -63,4 +68,181 @@ func init() {
...
+		for _, exception := range convertedSuppressions {
+			err := cli.OutputJSON(exception)
+			if err != nil {
+				return
+			}
+		}
+		colorizeR := color.New(color.FgRed, color.Bold)
+		cli.OutputHuman(colorizeR.Sprintf("WARNING: Before continuing, " +
+			"please thoroughly inspect the above exceptions to ensure they are valid and" +
+			" required. By continuing, you accept liability for any compliance violations" +
```
Comment (garyboyes) [inline cli/cmd/suppressions.go:119]: Are we gonna need to get Michael Moore involved for legal advice here 😆

File: cli/cmd/suppressions.go [hunk 2]
```diff
@@ -63,4 +68,181 @@ func init() {
...
+		if condition != "*" {
+			// It appears that for GCP, the resourceName field for policy exceptions is in fact expecting
+			// users to provider the full GCP resource_id.
+			// Example resourceId: //compute.googleapis.com/projects/gke-project-01-c8403ba1/zones/us-central1-a/instances/squid-proxy
+			// This was not the case for legacy suppressions and in most cases it's unlikely that the
+			// users will have provided this. Instead, we are more likely to have
+			// the resource name provided. To cover this scenario we prepend the resource name
+			// from the legacy suppression with "*/" to make it match the resource name while
+			// wildcarding the rest of the resourceId
+			condition = "*/" + condition
```
Comment (iok0) [inline cli/cmd/suppressions.go:199]: were you able to double check this syntax a dev env.?

**Review by garyboyes — COMMENTED** [review]
Looks good to me (at least with my v limited Go experience...)
Couple of questions around the functionality, when the user does the migration, does it remove the old suppressions? and, is this only going to be a 1 time operation, or could a customer/user run this multiple times? and what's the impact if they do that?

**Comment by iok0** [comment]
In terms of testing, was there an environment with a bunch of suppressions setup that you could try the migration out? i.e. check the expected exceptions were actually applied on next run?

File: cli/cmd/suppressions.go (hunk 2 above)
Comment (rmoles) [reply cli/cmd/suppressions.go:199]: Good point. I need to circle back an test this as I wasn't able to yesterday on dev5 or dev7

File: cli/cmd/suppressions.go [hunk 3]
```diff
@@ -63,4 +68,183 @@ func init() {
 	// gcp
 	suppressionsCommand.AddCommand(suppressionsGcpCmd)
 	suppressionsGcpCmd.AddCommand(suppressionsListGcpCmd)
+	suppressionsGcpCmd.AddCommand(suppressionsMigrateGcpCmd)
+}
+
+func autoConvertSuppressions(convertedPolicyExceptions []map[string]api.PolicyException) {
+	cli.StartProgress("Creating policy exceptions ...")
```
Comment (dmurray-lacework) [inline cli/cmd/suppressions.go:75]: Extreme nitpick, but I don't think we usually leave space between the ellipsis
```suggestion
	cli.StartProgress("Creating policy exceptions...")
```

File: cli/cmd/suppressions.go [hunk 4]
```diff
@@ -63,4 +68,183 @@ func init() {
...
+	for _, exceptionMap := range convertedPolicyExceptions {
+		for policyId, exception := range exceptionMap {
+			response, err := cli.LwApi.V2.Policy.Exceptions.Create(policyId, exception)
+			if err != nil {
+				cli.Log.Debug(err, "unable to create exception")
+				cli.OutputHuman(color.RedString(
+					"Error creating policy exception to create exception. %e"),
```
Comment (dmurray-lacework) [inline cli/cmd/suppressions.go:82]: Not sure this is the correct placeholder for your err.
https://pkg.go.dev/fmt
```suggestion
					"Error creating policy exception to create exception. %s"),
```

**Review by jon-stewart — APPROVED** [review]

File: cli/cmd/suppressions_gcp.go [hunk 5]
```diff
@@ -58,3 +156,213 @@ func suppressionsGcpList(_ *cobra.Command, _ []string) error {
...
+		if err != nil {
+			return err
+		}
+		if confirm {
+			autoConvertSuppressions(convertedPolicyExceptions)
+			printDiscardedSuppressions(discardedSuppressions)
+			cli.OutputHuman(color.GreenString("To view the newly created Exceptions, " +
+				"try running `lacework policy-exceptions list <policyId>"))
+		} else {
+			cli.OutputHuman("Cancelled Legacy Suppression to Exception migration!")
```
Comment (dmurray-lacework) [inline cli/cmd/suppressions_gcp.go:235]: nit, need a new line here to stop % being output
```suggestion
			cli.OutputHuman("Cancelled Legacy Suppression to Exception migration!\n")
```

**Review by dmurray-lacework — APPROVED** [review]

File: cli/cmd/suppressions.go (hunk 2 above)
Comment (rmoles) [reply cli/cmd/suppressions.go:199]: Confirmed on dev8 that the suppression format is working as expected. The only concern would be if a user has already provided the full resource ID. However I don't believe this format was used anywhere in the legacy compliance checks.

**Pull request merged**